datastore/datastore.go

```diff
@@ -417,6 +440,25 @@ func (s *FileStore) NameKey(kind, name string) *Key {
 	return &Key{Kind: kind, Name: name}
 }
 
+// IncompleteKey returns an incomplete key given a kind.
+func (s *FileStore) IncompleteKey(kind string) *Key {
+	// Continue selecting an ID until we find one not used.
```

Comment:
Could we use a UUID generator for this purpose (check out https://github.com/google/uuid) ? ISTM this algorithm is kind of clunky.